### ISSUE TITLE
Disconnect error

### ISSUE THREAD

**Issue by 2-bithacker** [deathstar/QServ2020#9]
I set it up for the first time and I got this error:
![Screenshot from 2022-03-17 11-42-40](https://user-images.githubusercontent.com/90713990/158850600-86ef8ff9-087b-43e6-8eb5-c799f643244c.png)

What is wrong?  What should I do to fix?

**Comment by deathstar** [comment]
Hmm that’s strange. It might have something to do with the GeoIP

**Comment by deathstar** [comment]
The old binaries are very buggy so that could be why

**Comment by 2-bithacker** [comment]
I built some new ones though

**Comment by deathstar** [comment]
Hmm. I have a couple servers up on a Ubuntu Linux server. Are you using Mac or Linux? If so what distro? Let me take a look when I get home. Presumably something to do with porting the mod to the new edition of sauer or the geoip since it’s on connect.

**Comment by 2-bithacker** [comment]
I am running it on a raspberry pi with debian 11.  I had to build some new binaries cuz raspberry pi can only do arm binaries.

Is there a way I can disable the geoip disconnect thingy?

**Comment by deathstar** [comment]
The message error could be caused by a localhost connect, sorry for the late reply.

**Comment by 2-bithacker** [comment]
that is the case

**Comment by deathstar** [comment]
Did you compile the latest source code or use an included binary?

**Comment by 2-bithacker** [comment]
compile. its a raspberry pi with a ARM cpu, so compiling is the only option.  The server works fine on a normal machine.  Seems it is a raspberry pi problem or something.

**Comment by 2-bithacker** [comment]
I did some more messing around, and it seems to work for a minute... It will load the map, and send me the join message, but then it promptly disconnects me. Says its a message error.

**Comment by deathstar** [comment]
I think it has to do with connecting from an internal ip address from local host Sent from my iPhoneOn Sep 20, 2023, at 3:38 PM, Fred Daws ***@***.***> wrote:﻿
I did some more messing around, and it seems to work for a minute... It will load the map, and send me the join message, but then it promptly disconnects me.

**Comment by deathstar** [comment]
Try connecting from external ip on server list Sent from my iPhoneOn Sep 20, 2023, at 8:29 PM, George Scott ***@***.***> wrote:﻿I think it has to do with connecting from an internal ip address from local host Sent from my iPhoneOn Sep 20, 2023, at 3:38 PM, Fred Daws ***@***.***> wrote:﻿
I did some more messing around, and it seems to work for a minute... It will load the map, and send me the join message, but then it promptly disconnects me.

**Comment by deathstar** [comment]
IE 66.215.103.152 instead of 192.168.1.1 you should also be able to untick “search local host” on the server list Sent from my iPhoneOn Sep 20, 2023, at 8:29 PM, George Scott ***@***.***> wrote:﻿I think it has to do with connecting from an internal ip address from local host Sent from my iPhoneOn Sep 20, 2023, at 3:38 PM, Fred Daws ***@***.***> wrote:﻿
I did some more messing around, and it seems to work for a minute... It will load the map, and send me the join message, but then it promptly disconnects me.

**Comment by 2-bithacker** [comment]
It is not a localhost exactly. Its another computer on my local network. search LAN is disabled, and it still gives me the error.

**Comment by 2-bithacker** [comment]
I hosted a zeromod server on that same machine im hosting the qserv one on before, and I can connect to it without issue.

**Comment by deathstar** [comment]
Yeah it's a problem with the way it checks the IPs to see if they're localhost so it doesn't use GeoIP on them. I am super busy but at one point it should be fixed. Can avoid by just not connecting from localhost ip from now (there should always be an external and internal server on your serverlist as long as you forward ports)

**Comment by 2-bithacker** [comment]
its not a localhost, its on a different machine on my network.
